File: backend/product-service/src/main/java/com/mall/product/entity/Comment.java
package com.mall.product.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("product_comment")
public class Comment {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long productId;
    private Long userId;
    private String content;
    private Integer rating;
    private String imageUrls;
    private LocalDateTime createdTime;
}

File: admin-frontend/src/api/comment.ts
import request from './request'

export interface CommentItem {
  id: number
  productId: number
  productName: string
  userId: number
  username: string
  content: string
  rating: number
  imageUrls: string[]
  createdTime: string
}

export function getAdminCommentList(params: Record<string, any>) {
  return request.get<any, CommentItem[]>('/product/admin/comment/list', { params })
}

export function deleteAdminComment(id: number) {
  return request.delete<void>(`/product/admin/comment/${id}`)
}
